console-subscriber/src/aggregator.rs

```diff
@@ -59,6 +61,12 @@ pub(crate) struct Aggregator {
 
     /// Map of task IDs to task stats.
     stats: TaskData<Stats>,
+
+    /// A counter for the pretty task IDs.
+    task_id_counter: TaskId,
+
+    /// A table that contains the span ID to pretty task ID mappings.
+    task_id_mappings: HashMap<span::Id, TaskId>,
```

Comment:
a possible optimization for this is adding a custom `Hasher` implementation that always just calls `span::Id::into_u64` and returns that value as the hash, rather than actually  hashing them. but, we can do that in a follow-up...especially because I don't really think the performance impact is particularly significant.